### ISSUE TITLE
azurerm_cosmosdb_account ip_range_filter doesn't actually allow CIDR

### ISSUE THREAD

**Issue by CryptonZylog** [terraform-providers/terraform-provider-azurerm#2146]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

>Terraform v0.11.8
>+ provider.azurerm v1.16.0
>+ provider.external v1.0.0
>+ provider.random v2.0.0

### Affected Resource(s)

* azurerm_cosmosdb_account

### Terraform Configuration Files

```hcl
variable "our_ip_whitelist" {
  default = ["0.0.0.0/24" ... ] # Example list of our public IPs
}

locals {
  
  # IP Range Filter here is to allow azure portal access
  cosmosdb_ip_range_azure = [
    "104.42.195.92/32",
    "40.76.54.131/32",
    "52.176.6.30/32",
    "52.169.50.45/32",
    "52.187.184.26/32"
  ]
}

resource "azurerm_cosmosdb_account" "nameredacted" {
  name                  = "${var.name}"
  resource_group_name   = "${var.resource_group_name}"
  location              = "${var.location}"
  tags                  = "${var.tags}"

  offer_type            = "${var.offer_type}"
  kind                  = "${var.kind}"

  # IPs https://docs.microsoft.com/en-us/azure/cosmos-db/firewall-support#connections-from-the-azure-portal
  ip_range_filter       = "${join(",",concat(local.cosmosdb_ip_range_azure,var.our_ip_whitelist)}"
  is_virtual_network_filter_enabled = "true"
  virtual_network_rule {
      id                = "${var.vnet_subnet_id}"
  }

  consistency_policy {
      consistency_level = "Session"
  }

  geo_location {
      location          = "${var.location}"
      failover_priority = 0
  } 
}
```

### Expected Behavior

IPs with CIDR notations added to CosmosDB firewall.

For sake of sanity:

```hcl
output "test" {
  value = "${join(",",concat(local.cosmosdb_ip_range_azure,var.our_ip_whitelist))}"
}
```

produces

`test = 104.42.195.92/32,40.76.54.131/32,52.176.6.30/32,52.169.50.45/32,52.187.184.26/32,0.0.0.0/24`

### Actual Behavior

`terraform plan`

```
Error: module.cosmosdb.azurerm_cosmosdb_account.sirenplatform: invalid value for ip_range_filter (Cosmos DB ip_range_filter must be a set of CIDR IP addresses separated by commas with no spaces: '10.0.0.1,10.0.0.2,10.20.0.0/16')
```

### Steps to Reproduce

See configuration example above, then run `terraform plan`

### Important Factoids

Actually adding IPs manually in Azure Portal, then trying to `terraform plan` to check sanity if the string is correct gives me this plan:

```
 ~ cosmosdb.azurerm_cosmosdb_account.<redacted>
      ip_range_filter: "0.0.0.0/24,1.1.1.1/24,104.42.195.92,40.76.54.131,52.176.6.30,52.169.50.45,52.187.184.26" => ""
```
-------------------

In short, it seems that validation for `ip_range_filter` is erroneous. All IPs provided are in CIDR have no spaces and are separated by commas.

**Comment by CryptonZylog** [comment]
Hm.. it seems that upon closer inspection of [regex](https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/resource_arm_cosmos_db_account.go#L77) for this field, it's not allowing a `/32` (single IP) even if Azure allows it. In fact neither /31, /30 work, but /29 does.
